bundles/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/handler/SamsungTvHandler.java

```diff
@@ -231,8 +234,9 @@ private synchronized void createService(RemoteDevice device) {
             SamsungTvService existingService = findServiceInstance(type);
 
             if (existingService == null || !existingService.isUpnp()) {
+                int port = ((Integer) getConfig(SamsungTvConfiguration.PORT)).intValue();
                 SamsungTvService newService = ServiceFactory.createService(type, upnpIOService, udn,
-                        configuration.refreshInterval, configuration.hostName, configuration.port);
+                        configuration.refreshInterval, configuration.hostName, port);
```

Comment:
Initial port has come from thing configuration, which seems to be updated by the service. With this change, port will always come from the service, could there be situation where service actually don't know the port and therefore the initial port should be used?

Reply:
The initial port from the thing configuration is used in case the port setting is set to something different from none. In this case, port and protocol are coming from the thing configuration.

The new ode is first searching the value in the local settings and if not found in the thing configuration. At the startup, the local settings is empty and so the thing configuration settings will be used. The local settings are only set when the handler is overriding a thing configuration setting.